sdk/core/Azure.Core/src/TokenCredential.cs

```diff
@@ -11,6 +12,13 @@ namespace Azure.Core
     /// </summary>
     public abstract class TokenCredential
     {
+        /// <summary>
+        /// Returns a <see cref="TimeSpan"/> value representing the amount of time to subtract from the token expiry time, whereupon
+        /// attempts will be made to refresh the token. By default this will occur two minutes prior to the expiry of the token.
+        /// </summary>
+        /// <returns>The duration value representing the amount of time to subtract from the token expiry time.</returns>
+        public virtual TokenRefreshOptions TokenRefreshOffset { get; } = new TokenRefreshOptions(TimeSpan.FromMinutes(2));
```

Comment:
Should this property name be `RefrestOptions`?

Reply:
Can't we control the same thing by changing ExpiresOn ? https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core/src/AccessToken.cs#L32

Reply:
No, we need the value of `ExpireOn` as well so that while one request refreshes the token the other requests can use the old one while it is valid.